### PR TITLE
Fixed Crash with RichTextLabel [fill] tag #51968 

### LINKED ISSUE #51968: Crash with RichTextLabel [fill] tag
### Godot version

3.x 16642e2

### System information

W10

### Issue description

```
[0] CowData<int>::get (C:\godot_source\core\cowdata.h:156)
[1] CowData<int>::get (C:\godot_source\core\cowdata.h:156)
[2] Vector<int>::operator[] (C:\godot_source\core\vector.h:87)
[3] RichTextLabel::_process_line (C:\godot_source\scene\gui\rich_text_label.cpp:452)
[4] RichTextLabel::_validate_line_caches (C:\godot_source\scene\gui\rich_text_label.cpp:1525)
[5] RichTextLabel::_notification (C:\godot_source\scene\gui\rich_text_label.cpp:996)
[6] RichTextLabel::_notificationv (C:\godot_source\scene\gui\rich_text_label.h:38)
[7] Object::notification (C:\godot_source\core\object.cpp:929)
[8] CanvasItem::_update_callback (C:\godot_source\scene\2d\canvas_item.cpp:445)
[9] MethodBind0<CanvasItem>::call (C:\godot_source\core\method_bind.gen.inc:149)
[10] Object::call (C:\godot_source\core\object.cpp:918)
[11] MessageQueue::_call_function (C:\godot_source\core\message_queue.cpp:241)
[12] MessageQueue::flush (C:\godot_source\core\message_queue.cpp:286)
[13] SceneTree::idle (C:\godot_source\scene\main\scene_tree.cpp:525)
[14] Main::iteration (C:\godot_source\main\main.cpp:2162)
[15] OS_Windows::run (C:\godot_source\platform\windows\os_windows.cpp:3307)
[16] widechar_main (C:\godot_source\platform\windows\godot_windows.cpp:161)
[17] _main (C:\godot_source\platform\windows\godot_windows.cpp:183)
[18] main (C:\godot_source\platform\windows\godot_windows.cpp:195)
[19] __scrt_common_main_seh (D:\a01\_work\26\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[20] BaseThreadInitThunk
-- END OF BACKTRACE --
```

The crash appeared in 1de8f31448518f3621ef4824546d481765ba9807

### Steps to reproduce

1. Add RichTextLabel
2. Enable bbcode
3. Type `[fill]whatever` in bbcode text field

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by anazr9**
Fixed [#51968](https://github.com/godotengine/godot/issues/51968) 
I based the changes off of 3.3.

### CODE REVIEW

File: scene/gui/rich_text_label.cpp [hunk 1]
```diff
@@ -447,7 +449,9 @@ int RichTextLabel::_process_line(ItemFrame *p_frame, const Vector2 &p_ofs, int &
 					fh = line_ascent + line_descent;
 
 					if (end && c[end - 1] == ' ') {
-						if (align == ALIGN_FILL) {
+						if (p_mode == PROCESS_CACHE) {
+							spaces_size += font->get_char_size(' ').width;
+						} else if (align == ALIGN_FILL) {
```
Comment (bruvzg) [inline scene/gui/rich_text_label.cpp:454]: What's the purpose for unused `spaces_size` variable?

```suggestion
						if (align == ALIGN_FILL && p_mode != PROCESS_CACHE) {
```
This should do  the same, but without unnecessary `get_chat_size` calls.

**Review by bruvzg — CHANGES_REQUESTED** [review]

File: scene/gui/rich_text_label.cpp (hunk 1 above)
Comment (anazr9) [reply scene/gui/rich_text_label.cpp:454]: I assumed it would be able to handle `ProcessMode p_mode` without raising errors.

**Review by bruvzg — DISMISSED** [review]
*Edit:* It was an issue with old code before #50483, not with this PR.

**Comment by bruvzg** [comment]
> it's breaking "fill" alignment.

Actually, it was broken before. It's fixing the crash caused by #50483, but I guess something else was already wrong.

**Review by bruvzg — APPROVED** [review]
PR is OK for fixing the crash (reverting the wrong part of #50483). ~but it's better to fix "fill" instead of reverting to the old broken code.~ Sorry, I was probably wrong, it's intended to work like this (only apply fill after the line break).

**Comment by bruvzg** [comment]
Ignore the previous comment about "fill", I was confused by the intended way "fill" should work, everything is OK as is.

**Pull request merged**